python/kserve/kserve/model.py

```diff
@@ -75,19 +89,43 @@ def __init__(self, name: str):
         self._http_client_instance = None
         self._grpc_client_stub = None
 
-    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR):
-        request = await self.preprocess(body) if inspect.iscoroutinefunction(self.preprocess) \
-            else self.preprocess(body)
+    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR, headers={}):
+        request_id = headers.get("X-RequestID")
+
+        # latency vars
+        preprocess_ms = 0
+        explain_ms = 0
+        predict_ms = 0
+        postprocess_ms = 0
+
+        with PRE_HIST_TIME.time():
```

Comment:
each step uses prometheus's `.time()` to record a histogram, and python's time func for the log at the end that spits out just the latencies for each step per request id.